dev-libs/libprelude/files/libprelude-4.0.0-fix-python-bindings.patch

```diff
@@ -0,0 +1,65 @@
+--- a/bindings/python/Makefile.am	2017-07-17 12:03:05.653000000 +0200
++++ b/bindings/python/Makefile.am	2017-09-06 18:51:57.179778549 +0200
+@@ -9,35 +9,6 @@
+ all-am: python-build
+ 
+ python-build: _prelude.cxx
+-if HAVE_PYTHON2
+-	CC="$(CXX)" $(PYTHON2) setup.py build
+-endif
+-if HAVE_PYTHON3
+-	CC="$(CXX)" $(PYTHON3) setup.py build
+-endif
+-install-exec-hook:
+-if HAVE_PYTHON2
+-	$(PYTHON2) setup.py install `test -n "$(DESTDIR)" && echo --root $(DESTDIR)` --prefix @prefix@
+-endif
+-if HAVE_PYTHON3
+-	$(PYTHON3) setup.py install `test -n "$(DESTDIR)" && echo --root $(DESTDIR)` --prefix @prefix@
+-endif
+-
+-uninstall-hook:
+-if HAVE_PYTHON2
+-	$(PYTHON2) setup.py uninstall `test -n "$(DESTDIR)" && echo --root $(DESTDIR)` --prefix @prefix@
+-endif
+-if HAVE_PYTHON3
+-	$(PYTHON3) setup.py uninstall `test -n "$(DESTDIR)" && echo --root $(DESTDIR)` --prefix @prefix@
+-endif
+-
+-clean-local:
+-if HAVE_PYTHON2
+-	$(PYTHON2) setup.py clean -a
+-endif
+-if HAVE_PYTHON3
+-	$(PYTHON3) setup.py clean -a
+-endif
+ 
+ _prelude.cxx prelude.py: $(top_srcdir)/bindings/c++/include/*.hxx $(top_srcdir)/bindings/libpreludecpp.i libpreludecpp-python.i 
+ if HAVE_SWIG
+--- a/configure.ac	2017-07-17 12:03:05.872000000 +0200
```

Comment:
Why are you modifying this? It should not be necessary.

If it fails because only python2/python3 is found, then use `python_is_python3` to pass the appropriate option. AFAICS it also tries to guess versions directly; use the third argument of `use_with` to pass the correct value, i.e. something like:

    if python_is_python3; then
        myconf+=( --without-python2 --with-python3="${EPYTHON}" )
    else
        myconf+=( --without-python3 --with-python2="${EPYTHON}" )
    fi

(it's usually a good idea to put all configure flags in `myconf` array when the construction gets complex and call `econf "${myargs[@]}"` afterwards)

Reply:
I may miss something ...
With your suggestion, I can't have python2 AND python3, am I right ? Because I want to be able to have python2 and python3 support if it is asked.

If I don't patch configure.ac I can't find out how to set the right configure flags without using my first proposition on this.

Reply:
The configure script doesn't make a difference between python2/python3. You could pass both disguised as python2 and it will work as well.

Reply:
But what you propose has to be in a python_configure like function else EPYTHON is not defined.
right ?
So for each python, we will add these configure flags.

Reply:
OK, I have to add python_setup before